pkg/apis/pipeline/v1alpha1/pipeline_validation.go

```diff
@@ -124,6 +125,15 @@ func (ps *PipelineSpec) Validate(ctx context.Context) *apis.FieldError {
 	// Names cannot be duplicated
 	taskNames := map[string]struct{}{}
 	for _, t := range ps.Tasks {
+		// Task names are appended to the container name, which must exist and
+		// must be a valid k8s name
+		if errSlice := validation.IsQualifiedName(t.Name); len(errSlice) != 0 {
+			return apis.ErrInvalidValue(errSlice[0], "spec.tasks.name")
```

Comment:
Looking at https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go, it seems that it may be nice to get the whole error slice, instead of just errSlice[0].

Reply:
From what I recall, it is possible for two error strings to be returned where these could be either the name being too long or empty and the other being an invalid name regex. More than likely, there will only be one error so for the sake of formatting I did it this way. I could return the whole slice and string join it as well.

Reply:
Sure, I don't think it's worth holding this one on that, but I'd rather err on the side of giving more information in case of error :) It can also be enhanced in a follow-up.

Reply:
thanks for the update!